File: Heap/row-min-cost.java
package Heap;

import java.util.*;
import java.util.PriorityQueue;

class Main {

    public int minCost(int[] arr) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int cost = 0;
        for (int ele : arr) {
            pq.add(ele);
        }
        
        while (pq.size()>1) {
            int x = pq.remove();
            int y = pq.remove();
            int sum = x + y;
            cost += sum;
            pq.add(sum);
        }   
        return cost;
    }
    public static void main(String[] args) {
        int[] arr = {2,7,1,4,8};
        Main s = new Main();
        System.out.println(s.minCost(arr));
    }
}

File: Heap/row-min-cost.js
let minCost = (arr) => {
    let cost = 0;
    while (arr.length>1) {
        arr.sort((a,b)=>a-b);
        let x = arr.shift();
        let y = arr.shift();
        let sum = x+y;
        cost += sum;
        arr.push(sum);
    }
    return cost;
}

let arr = [2,7,1,4,8];

console.log(minCost(arr));